### PR TITLE
CI/github: add work-around for brew breakage on macOS

### PR DESCRIPTION
**Pull request by bagder**
... to fix the broken github CI jobs on macOS.

### CODE REVIEW

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -63,6 +63,9 @@ jobs:
     - run: echo libtool autoconf automake pkg-config ${{ matrix.build.install }} | xargs -Ix -n1 echo brew '"x"' > /tmp/Brewfile
       name: 'brew bundle'
 
+    - run: brew uninstall openssl@1.0.2t && brew uninstall python@2.7.17 && brew untap local/openssl && brew untap local/python2 && brew install openssl@1.1
```
Comment (mback2k) [inline .github/workflows/macos.yml:66]: Isn't this redundant to the version specified in install above?

File: .github/workflows/macos.yml [hunk 2]
```diff
@@ -101,15 +104,18 @@ jobs:
...
+          install: nghttp2 openssl@1.1
+          generate: -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl@1.1 -DCURL_DISABLE_LDAP=ON -DCURL_DISABLE_LDAPS=ON
         - name: LibreSSL
           install: nghttp2 libressl
           generate: -DOPENSSL_ROOT_DIR=/usr/local/opt/libressl -DCURL_DISABLE_LDAP=ON -DCURL_DISABLE_LDAPS=ON
     steps:
     - run: echo libtool autoconf automake pkg-config ${{ matrix.build.install }} | xargs -Ix -n1 echo brew '"x"' > /tmp/Brewfile
       name: 'brew bundle'
 
+    - run: brew uninstall openssl@1.0.2t && brew uninstall python@2.7.17 && brew untap local/openssl && brew untap local/python2
```
Comment (mback2k) [inline .github/workflows/macos.yml:116]: See above comment.

File: .github/workflows/macos.yml (hunk 1 above)
Comment (bagder) [reply .github/workflows/macos.yml:66]: redundant how? If you can improve this, please be my guest. It took me almost a dozen iterations to get this to work yday... =)

Comment (mback2k) [reply .github/workflows/macos.yml:66]: I will give it a try, it _should_ be possible to achieve this from within the generated Brewfile.

**Comment by bagder** [comment]
According to https://github.com/actions/virtual-environments/issues/1811, this will be fixed in "3-4 days" so ideally we can just back out the work-around again then?

**Comment by mback2k** [comment]
Good idea, let's see if it will be fixed by next week then.

**Pull request closed without merging**